src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/JndiDirectoryDAO.scala

```diff
@@ -54,8 +54,13 @@ class JndiDirectoryDAO(protected val directoryConfig: DirectoryConfig)(implicit
     }
   }
 
-  override def deleteGroup(groupName: WorkbenchGroupName): Future[Unit] = withContext { ctx =>
-    ctx.unbind(groupDn(groupName))
+  override def deleteGroup(groupName: WorkbenchGroupName): Future[Unit] = {
+    listAncestorGroups(groupName).flatMap { ancestors =>
+      if (ancestors.nonEmpty)
+        throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"group ${groupName.value} cannot be deleted because it is a member of at least 1 other group"))
```

Comment:
I don't think it is functionally different but you probably want to return Future.failed instead of throwing the exception.

Reply:
Cool, good to know.  FWIW:  https://stackoverflow.com/questions/31194138/difference-between-returning-future-failedexception-and-throwing-an-exception

Reply:
The discussion on the answer is pretty interesting, in particular https://stackoverflow.com/questions/31194138/difference-between-returning-future-failedexception-and-throwing-an-exception#comment50481668_31195611